### ISSUE TITLE
ChainerX does not have APIs to run backward of BatchNorm/Pooling without running forward

### ISSUE THREAD

**Issue by shinh** [chainer/chainer#6162]
Currently, BatchNorm and Pooling functions are implemented by `Device::Get*ForwardBackward` functions. I think this is a good abstraction to let devices decide which values should be retained. However, this interface prevents us from conducting experiments which need to run backward computations without running forward computations. For example, it's impossible to run backward computation in a different machine from the machine where forward computation ran. It would be great if there are interfaces with which users can directly run backward computation alone, though the users needs to manage retained Arrays by themselves.

On the other hand, Conv is OK as `ConvTranspose` is backprop of image side and we can use `Device::ConvGradWeight` for backprop of weight side.

**Comment by hvy** [comment]
Is this issue solved with the recently introduced `*GradKernel`s? E.g. https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/kernels/pooling.h#L108.

**Comment by hvy** [comment]
Closing this issue as fixed, although the state interface might be suboptimal.
